plugins/base/src/main/kotlin/parsers/Parser.kt

```diff
@@ -49,10 +51,49 @@ abstract class Parser {
 
     private fun jkdocToListOfPairs(javadoc: String): List<Pair<String, String>> =
         "description $javadoc"
-            .split("\n@")
+            .splitIgnoredInsideBackticks("\n@")
```

Comment:
A bit concerned about performance degradation after this change. Looking at the implementation of `split` and at yours, there's additional overhead added for each j/kdoc that's parsed, whereas by far not every doc comment contains code sections and may even have no tags (in this situation, split would do a lookup of the delimiter and just return the string if nothing is found, whereas `splitIgnoredInsideBackticks` would go char by char and copy it to the StringBuilder)

So it seems like this case is only relevant for comments where it contains both `@` symbol and backticks. I suggest we use the special function only if this condition holds true. Ideally, we'd want to check if there's at least one `@` is inside backticks (could be done via `indexOf` comparison), but I think it may be too much. Other than that, I think `contains` would call `nativeIndexOf` and it should be pretty fast.

```kotlin
private fun jkdocToListOfPairs(doc: String): List<Pair<String, String>> {
    val descriptionDoc = "description $doc"

    // annotations inside code blocks must not be regarded as kdoc/javadoc tags
    val mayContainAnnotationsInsideCodeSection = doc.contains("```") && doc.contains("\n@")
    val docTokens = if (mayContainAnnotationsInsideCodeSection) {
        descriptionDoc.splitIgnoredInsideBackticks()
    } else {
        descriptionDoc.split("\n@")
    }

    return docTokens.map { content ->
        val contentWithEscapedAts = content.replace("\\@", "@")
        val (tag, body) = contentWithEscapedAts.split(" ", limit = 2)
        tag to body
    }
}
```

Reply:
This is a good point, but I have another solution.

Reply:
It's definitely better, but I still think it's overkill to migrate to this function exclusively for such a rare corner case. I believe Kotlin's `split` is better tested and outperforms `splitIgnoredInsideBackticks` for the majority of cases.

Any objections to what I proposed above about using `splitIgnoredInsideBackticks` only when backticks and `@` are present?

Reply:
1. I am not sure It is a rare corner case. See the attached issues.
2. I am not sure `split` outperforms `splitIgnoredInsideBackticks`. See `split` implementation in JVM. Anyway there are cases when we should use `splitIgnoredInsideBacktick`. It have to be tested as well.
3. The suggested code have 2 redundant  invocations of `contains` with linear complexity. 
4. This is a heuristic that make the code more complicated to understand a normal flow.

Reply:
Discussed it in DMs. This method is only used for `module.md` files, so it's not as performance critical as I thought it was. I'll suggest a few renames, but other than that I think it'll be fine